RulerDetectionPlugin/src/RulerDetectionPlugin.cpp

```diff
@@ -0,0 +1,611 @@
+/*******************************************************************************************************
+ RulerDetectionPlugin.cpp
+
+ nomacs is a fast and small image viewer with the capability of synchronizing multiple instances
+
+ Copyright (C) 2017 Pascal Plank
+
+ This file is part of nomacs.
+
+ nomacs is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ nomacs is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ *******************************************************************************************************/
+
+#include "RulerDetectionPlugin.h"
+
+
+#pragma warning(push, 0)	// no warnings from includes - begin
+#include <QAction>
+#include <QUuid>
+#include <QVector2D>
+#include <QSharedPointer>
+#include <qfiledialog.h>
+#include <qmessagebox.h>
+
+#include "opencv2/xfeatures2d/nonfree.hpp"
+#include "opencv2\opencv.hpp"
```

Comment:
accurately add opencv modules (e.g. `#include <opencv2/calib3d/calib3d.hpp>` for homography)